### PR TITLE
OI LED and LCD support

### PR DESCRIPTION
**Pull request by jonahb55**

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -319,10 +335,12 @@ private void configureInputs() {
...
     operatorOI.getShooterRollerButton()
         .whileActiveContinuous(new RunShooterRoller(shooterRoller).alongWith(new RunHopper(hopper)));
-    operatorOI.getShooterUnstickButton().whileActiveContinuous(new FeedUnstick(shooterRoller, hopper));
+    operatorOI.getShooterUnstickButton()
+        .whileActiveContinuous(new FeedUnstick(shooterRoller, hopper, operatorOI::updateLED));
 
-    operatorOI.getIntakeExtendButton().whenActive(new InstantCommand(intake::extend, intake));
-    operatorOI.getIntakeRetractButton().whenActive(new InstantCommand(intake::retract, intake));
+    operatorOI.getIntakeExtendButton().whenActive(() -> intake.extend());
+    operatorOI.getIntakeRetractButton().whenActive(() -> intake.retract());
```
Comment (e-bonner) [inline src/main/java/frc/robot/RobotContainer.java]:
```suggestion
    operatorOI.getIntakeExtendButton().whenActive(intake::extend, intake);
    operatorOI.getIntakeRetractButton().whenActive(intake::retract, intake);
```
These should require the intake (and method references are a little cleaner than lambdas here)

File: src/main/java/frc/robot/commands/SetLEDOverride.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import frc.robot.oi.IOperatorOI.OILED;
+import frc.robot.oi.IOperatorOI.OILEDState;
+import frc.robot.util.UpdateLEDInterface;
+
+// NOTE:  Consider using this command inline, rather than writing a subclass.  For more
+// information, see:
+// https://docs.wpilib.org/en/latest/docs/software/commandbased/convenience-features.html
+public class SetLEDOverride extends InstantCommand {
```
Comment (e-bonner) [inline src/main/java/frc/robot/commands/SetLEDOverride.java]: Why do we need this command? In RobotContainer we can do
`button.whenActive(() -> operatorOI.updateLED(OILED, OILEDState))`

File: src/main/java/frc/robot/oi/OIArduinoConsole.java [hunk 3]
```diff
@@ -41,6 +50,16 @@
...
 
+    NetworkTable ledTable;
+    NetworkTableEntry ledEntry;
+
+    OILCDField LCDTimer;
+    OILCDField LCDPressure;
+    OILCDField LCDFlyWheelSpeed;
+    OILCDField LCDHoodPosition;
+
+    private static final Map<OILED, Integer> ledMap = new HashMap<OILED, Integer>();
```
Comment (e-bonner) [inline src/main/java/frc/robot/oi/OIArduinoConsole.java]: If this is fixed we should use a fixed map. See `Map.of()`.

File: src/main/java/frc/robot/oi/OIeStopConsole.java [hunk 4]
```diff
@@ -46,6 +49,8 @@
   NetworkTable ledTable;
   NetworkTableEntry ledEntry;
 
+  private static final Map<OILED, Integer> ledMap = new HashMap<OILED, Integer>();
```
Comment (e-bonner) [inline src/main/java/frc/robot/oi/OIeStopConsole.java]: Same map comment as the Arduino console.

File: src/main/java/frc/robot/subsystems/ShooterHood.java [hunk 5]
```diff
@@ -63,4 +72,20 @@ public void setStopPosition(boolean stopped) {
       }
     }
   }
+
+  /**
+   * Updates hood position LEDs
+   * 
+   * @param led LED to illuminate
+   */
+  public void setLEDs(HoodPosition position) {
```
Comment (e-bonner) [inline src/main/java/frc/robot/subsystems/ShooterHood.java]: Why is this in the subsystem if it doesn't integrate with anything else? Why not just have the command set the LEDs?

File: src/main/java/frc/robot/util/OILCDField.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.util;
```
Comment (e-bonner) [inline src/main/java/frc/robot/util/OILCDField.java]: Why is this in util? Shouldn't it be in ArduinoConsole?

File: src/main/java/frc/robot/util/PressureSensor.java [hunk 7]
```diff
@@ -51,7 +53,7 @@ public double getVoltage() {
     @Override
     public void periodic() {
         if (available()) {
-            SmartDashboard.putNumber("Pressure Sensor", getPressure());
```
Comment (e-bonner) [inline src/main/java/frc/robot/util/PressureSensor.java]: I would keep the SD code.

File: src/main/java/frc/robot/util/SetFlyWheelSpeedInterface.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.util;
```
Comment (e-bonner) [inline src/main/java/frc/robot/util/SetFlyWheelSpeedInterface.java]: Why util and not IOperatorOI where the methods are?

File: src/main/java/frc/robot/util/SetHoodPositionLCDInterface.java [hunk 9]
```diff
@@ -0,0 +1,18 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.util;
```
Comment (e-bonner) [inline src/main/java/frc/robot/util/SetHoodPositionLCDInterface.java]: Why not IOperatorOI?

File: src/main/java/frc/robot/util/SetPressureInterface.java [hunk 10]
```diff
@@ -0,0 +1,16 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.util;
```
Comment (e-bonner) [inline src/main/java/frc/robot/util/SetPressureInterface.java]: See the other LCD interface comments.

File: src/main/java/frc/robot/util/UpdateLEDInterface.java [hunk 11]
```diff
@@ -0,0 +1,19 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.util;
```
Comment (e-bonner) [inline src/main/java/frc/robot/util/UpdateLEDInterface.java]: See previous file comments.

File: src/main/java/frc/robot/commands/SetLEDOverride.java (hunk 2 above)
Comment (jonahb55) [reply src/main/java/frc/robot/commands/SetLEDOverride.java]: When we tried that, it didn't seem to run when disabled. Is there an alternate solution?

**Pull request merged**